app/src/main/java/helium314/keyboard/latin/LatinIME.java

```diff
@@ -1613,6 +1614,17 @@ public void getSuggestedWords(final int inputStyle, final int sequenceNumber,
                 mKeyboardSwitcher.getKeyboardShiftMode(), inputStyle, sequenceNumber, callback);
     }
 
+    public void showToast(final int resId, final int timeMillis, final boolean fallback){
+        // In API 32 and below, toasts can be shown without a notification permission.
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.S_V2){
+            final Toast toast = Toast.makeText(this, resId, timeMillis);
```

Comment:
Does this really work?
According to [documentation](https://developer.android.com/reference/android/widget/Toast#makeText(android.content.Context,%20java.lang.CharSequence,%20int)) this is wrong.

Reply:
it works, not sure why the documents states it must be one of those 2 if it is not enforced, but I will change it anyway to not risk it. Thanks for noticing

Reply:
It works because Android only checks whether the number is `Toast.LENGTH_LONG`: https://android.googlesource.com/platform/frameworks/base/+/f4bed684c939b0f8809ef404b8609fe4ef849263/core/java/android/widget/Toast.java#452